common/src/primitives/encoding/bech32m.rs

```diff
@@ -0,0 +1,52 @@
+use super::Bech32Error;
+use super::DecodedBase32FromBech32;
+use bech32::CheckBase32;
+use bech32::{self, Variant};
+
+pub fn base32_to_bech32m<T: AsRef<[u8]>>(hrp: &str, data: T) -> Result<String, Bech32Error> {
+    let data = data.check_base32()?;
+    bech32::encode(hrp, data, Variant::Bech32m).map_err(|e| e.into())
+}
+
+#[allow(dead_code)]
+pub fn bech32m_to_base32(s: &str) -> Result<DecodedBase32FromBech32, Bech32Error> {
+    match bech32::decode(s) {
+        Ok((hrp, base32, variant)) => {
+            if variant == Variant::Bech32 {
+                return Err(Bech32Error::UnsupportedVariant);
+            }
+
+            // ------- this checking is only for BITCOIN: Witness Programs
+            // if hrp == "bc" && ( s.len() < 2 || s.len() > 40 ) {
+            //     return Err(Bech32Error::InvalidLength);
+            // }
+            // ------- EOL
+
+            let data = base32.iter().map(|x| x.to_u8()).collect();
```

Comment:
This is a crime. There's no relation between a u5 array and a u8 array without a proper bit conversion. This is like convert a binary 1100110 to a decimal 1100110.

Reply:
Gotcha! Fixed!